backend/src/controllers/reports/rpt_PA_Fiscal_Registry.js

```diff
@@ -0,0 +1,93 @@
+const useCommonComponents = require("../useCommonComponents/index");
+const useController = require("../useController/index");
+const model = require("@models/reports/rpt_PA_Fiscal_Registry");
+const utils = require("./helpers");
+const what = { single: "report", plural: "reports" };
+const controller = useController(model, what);
+const _ = require("lodash");
+
+// Template and data reading
+const cdogs = useCommonComponents("cdogs");
+const { getReport, getDocumentApiBody, pdfConfig, groupByProperty } = utils;
+
+controller.getReport = getReport;
+
+/**
+ * Get a Project rollup Report for a specific array of portfolio.
+ *
+ * @param   {FastifyRequest} request FastifyRequest is an instance of the standard http or http2 request objects.
+ * @param   {FastifyReply}   reply   FastifyReply is an instance of the standard http or http2 reply types.
+ * @returns {object}
+ */
+controller.rpt_PA_Fiscal_Registry = async (request, reply) => {
+  // const portfolios = [
+  //   {
+  //     portfolio:"citz",
+  //     projectData: [
+  //       {
+  //         portfolio: "citz",
+  //         projectName: "my project name",
+  //       },
+  //       {
+  //         portfolio: "citz",
+  //         projectName: "my project name",
+  //       },
+  //     ],
+  //     totalBudget: "1,022,000",
+  //   },
+  //   {
+  //     portfolio:"test2",
+  //     projectData: [
+  //       {
+  //         projectName: "my project name2",
+  //       },
+  //       {
+  //         projectName: "my project name2",
+  //       },
+  //     ],
+  //     totalBudget: "3,442,030",
+  //   },
+  // ];
+
+  controller.userRequires(request, what, "reports_read_all");
+  try {
+    const fiscalRegistry = await model.fiscalRegistry(request.query);
+    const plannedBudgetTotals = await model.planned_budget_totals(request.query);
+    const reportTotal = await model.report_total(request.query);
+
+    const fiscalRegistryGroupedByPortfolioName = groupByProperty(fiscalRegistry, "portfolio_name");
+    const plannedBudgetTotalsKeyedByPortfolioId = _.keyBy(plannedBudgetTotals, "portfolio_name");
+
+    const fiscalRegistryBudgetTotals = _.map(fiscalRegistryGroupedByPortfolioName, (portfolio) => ({
+      ...portfolio,
+      total_budget: plannedBudgetTotalsKeyedByPortfolioId[portfolio.portfolio_name].total_budget,
+    }));
+
+    const reportTotalRow = _.first(reportTotal).total_budget;
```

Comment:
I wonder if it would be better to save defining two variables b efore use, and instead simply do:
```javascript
const result = {
  fiscalRegistry: _.map(fiscalRegistryGroupedByPortfolioName, (portfolio) => ({
      ...portfolio,
      total_budget: plannedBudgetTotalsKeyedByPortfolioId[portfolio.portfolio_name].total_budget,
    })),
  total: _.first(reportTotal).total_budget
  };
  ```